app/src/main/java/cm/aptoide/pt/analytics/FirstLaunchAnalytics.java

```diff
@@ -334,12 +327,23 @@ private void setUTMDimensionsToUnknown() {
     FlurryAgent.addSessionProperty(UTM_CAMPAIGN, UNKNOWN);
     FlurryAgent.addSessionProperty(UTM_CONTENT, UNKNOWN);
     FlurryAgent.addSessionProperty(ENTRY_POINT, UNKNOWN);
+  }
+
+  private void sendRakamUserProperties(String utmContent, String utmSource, String utmCampaign,
+      String utmMedium, String entryPoint, String packageName) {
+    Identify rakamProperties = new Identify();
+
+    rakamProperties.set(GMS_RAKAM, rakamGmsValueCache);
```

Comment:
rakamGmsValueCache might be null. It depends in another method being called, which is called in AnalyticsActivity... this should be reviewed.

Reply:
Created a wrapper that will get the value from the adnetworks (regarding gms). Started using it the first time i got it and will calculate it if it does not exist. I am using that "singleton" on both places.